src/js/tracks/track.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * @file track.js
+ */
+import * as browser from '../utils/browser.js';
+import document from 'global/document';
+import * as Guid from '../utils/guid.js';
+import EventTarget from '../event-target';
+
+/**
+ * setup the common parts of an audio, video, or text track
+ * @link https://html.spec.whatwg.org/multipage/embedded-content.html
+ *
+ * @param {String} type The type of track we are dealing with audio|video|text
+ * @param {Object=} options Object of option names and values
+ * @extends EventTarget
+ * @class Track
+ */
+class Track extends EventTarget {
+  constructor(options = {}) {
+    super();
+    if (!options.tech) {
+      throw new Error('A tech was not provided.');
+    }
+
+    let track = this;
+    let type = options.trackType || 'generic';
+    if (browser.IS_IE8) {
+      track = document.createElement('custom_' + type);
+
+      for (let prop in track.prototype) {
+        if (prop !== 'constructor') {
+          track[prop] = track.prototype[prop];
+        }
+      }
+    }
+    track.tech_ = options.tech;
+
+    let trackProps = {
+      id: options.id || 'vjs_' + type + '_track_' + Guid.newGUID(),
+      kind: options.kind || '',
+      label: options.label || '',
+      language: options.language || ''
+    };
+
+    for(let key in trackProps) {
+      Object.defineProperty(track, key, {
+        get() { return trackProps[key]; },
+        set() {}
```

Comment:
it seems that removing the noop causes trying to set these variables to throw an error. Is that what we want to happen?